probe-rs/src/probe/jlink/mod.rs

```diff
@@ -335,36 +340,117 @@ impl DebugProbe for JLink {
+                let jtag_to_swd_sequence = [
+                    false, true, true, true, true, false, false, true, true, true, true, false,
+                    false, true, true, true,
+                ];
+
+                // Construct the entire init sequence.
+                let swd_io_sequence =
+                    // Send the reset sequence (> 50 0-bits).    
+                    iter::repeat(true).take(64)
+                    // Send the JTAG to SWD sequence.
+                    .chain(jtag_to_swd_sequence.iter().copied())
+                    // Send the reset sequence again in case we were in SWD mode already (> 50 0-bits).
+                    .chain(iter::repeat(true).take(64))
+                    // Send 10 idle line bits.
+                    .chain(iter::repeat(false).take(10));
+
+                // Construct the direction sequence for reset sequence.
+                let direction =
+                    // Send the reset sequence (> 50 0-bits).    
+                    iter::repeat(true).take(64)
+                    // Send the JTAG to SWD sequence.
+                    .chain(iter::repeat(true).take(16))
+                    // Send the reset sequence again in case we were in SWD mode already (> 50 0-bits).
+                    .chain(iter::repeat(true).take(64))
+                    // Send 10 idle line bits.
+                    .chain(iter::repeat(true).take(10));
+
+                // Send the init sequence.
+                // We don't actually care about the response here.
+                // A read on the DPIDR will finalize the init procedure and tell us if it worked.
+                jlink.swd_io(direction, swd_io_sequence)?;
+
+                // So we read the DPIDR to complete the init sequence.
+                let response = DAPAccess::read_register(self, PortType::DebugPort, 0x0000)?;
+
+                // If we get to here, the init worked.
+                let dpidr = DPIDR(response);
+                log::debug!("Successfully read DPIDR. SWD init successful.");
+                log::debug!(
+                    "Manufacturer is {} with part_no={} and revision={}",
+                    jep106::JEP106Code::new(dpidr.jep_cc(), dpidr.jep_id())
+                        .get()
+                        .unwrap_or("Unknown JEP106 code."),
+                    dpidr.part_no(),
+                    dpidr.revision(),
+                );
+
+                // Clear all error flags in the abort register.
+                let mut abort = Abort::default();
+                abort.set_orunerrclr(true);
+                abort.set_wderrclr(true);
+                DAPAccess::write_register(
+                    self,
+                    PortType::DebugPort,
+                    Abort::ADDRESS as u16,
+                    abort.into(),
+                )?;
+
+                // Enable & power up all debug facilities.
+                let mut ctrl = Ctrl::default();
+                ctrl.set_cdbgpwrupack(true);
```

Comment:
I think we should also set the `csyspwrupreq` bit here. 

The `cdbgpwrupack` bit should be checked to verify that the power up request was succesfull, it should not be written.

Reply:
Yes, checking is definitely a good idea. now I understand why it didn't have a setter -.- All the material on the internet kinda suggested setting that bit ... but maybe I just read wrong ... will fix this!